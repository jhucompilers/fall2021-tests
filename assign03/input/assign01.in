PROGRAM assign01;
    TYPE R = RECORD x, y: INTEGER; END;
    TYPE Q = RECORD a, b: INTEGER; END;

    VAR a: ARRAY 5 OF R;
    VAR b: ARRAY 5 OF Q;
BEGIN
    -- a := b;
    b[0].a := 1;
    b[0].b := 2;
    a[0].x := b[0].a;
    a[0].y := b[0].b;
END.
