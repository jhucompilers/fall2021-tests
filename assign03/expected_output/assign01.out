1,VAR,x,INTEGER
1,VAR,y,INTEGER
0,TYPE,R,RECORD (INTEGER x INTEGER)
1,VAR,a,INTEGER
1,VAR,b,INTEGER
0,TYPE,Q,RECORD (INTEGER x INTEGER)
0,VAR,a,ARRAY 5 OF RECORD (INTEGER x INTEGER)
0,VAR,b,ARRAY 5 OF RECORD (INTEGER x INTEGER)
